core/store/src/trie/prefetching_trie_storage.rs

```diff
@@ -469,6 +485,36 @@ fn prefetch_state_matches(expected: PrefetchSlot, actual: &PrefetchSlot) -> bool
     }
 }
 
+/// Guard that owns the spawned prefetching IO threads.
+#[must_use = "When dropping this handle, the IO threads will be aborted immediately."]
+pub(crate) struct PrefetchingThreadsHandle {
+    /// Shutdown channels to all spawned threads.
+    shutdown_channels: Vec<crossbeam::channel::Sender<()>>,
+    /// Join handles of spawned threads.
+    ///
+    /// Used to actively join all background threads after shutting them down.
+    handles: Vec<thread::JoinHandle<()>>,
+}
+
+impl Drop for PrefetchingThreadsHandle {
+    fn drop(&mut self) {
+        for tx in &self.shutdown_channels {
+            let e = tx.send(());
+            if e.is_err() {
+                // Usually senders are dropped after joining all background threads.
+                // But if this order is reversed, this send here will fail. This
+                // is perfectly valid behavior and should not be treated as error.
+                debug!("IO thread already hung up when trying to shut it down.");
+            }
+        }
```

Comment:
```suggestion
        self.shutdown_channels.clear()
```

Reply:
That is, closing a channel gives the signal we want here, we don't need special code to handle this.